datafusion_iceberg/src/table.rs

```diff
@@ -1643,4 +1668,13 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_fake_object_store_url() {
+        assert_eq!(
+            fake_object_store_url("s3://aaa/bbb/ccc"),
+            Some(ObjectStoreUrl::parse("s3://aaa-bbb-ccc").unwrap()),
```

Comment:
What about something like `s3://bucket/my-table` and `s3://bucket/my/table`, will these two collide?

Reply:
You're right, that would collide

Reply:
Updated. I hate what I've done but should now be collision free

Reply:
Might be nicer with just a `.replace('-', "--")` first to path replacements in the original version?

Reply:
@gruuya sorry, I'm not sure what you mean

Reply:
Basically this
```rust
fn fake_object_store_url(table_location_url: &str) -> Option<ObjectStoreUrl> {
    let mut u = url::Url::parse(table_location_url).ok()?;
    u.set_host(Some(&format!(
        "{}{}",
        u.host_str().unwrap_or("-"),
        u.path()
            .replace('-', "--") // ensure we escape existing hyphens by double hyphens
            .replace(object_store::path::DELIMITER, "-")
            .replace(':', "-")
    )))
    .unwrap();
    u.set_path("");
    u.set_query(None);
    u.set_fragment(None);
    ObjectStoreUrl::parse(&u).ok()
}


```

Reply:
I don't think that works, e.g. `-/--` would clash with `--/-`.

Reply:
I've simplified to hex encoding since we don't care about length